### ISSUE TITLE
TensorRT inference error: TensorFlow device was not registered

### ISSUE THREAD

**Issue by k-anastasia** [tensorflow/tensorflow#20780]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**:
Yes
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
Ubuntu 16.04
- **TensorFlow installed from (source or binary)**:
source
- **TensorFlow version (use command below)**:
last commit from pull request #20350 
- **Python version**: 
2.7
- **Bazel version (if compiling from source)**:
0.15.0
- **GCC/Compiler version (if compiling from source)**:
gcc 5.4.0
- **CUDA/cuDNN version**:
Cuda 9.0/cuDNN 7
- **GPU model and memory**:
2 x Tesla P40 (22919MiB)
- **Exact command to reproduce**:

### Describe the problem
Using tensorflow.contrib.tensorrt.create_inference_graph results in an error:
`Non-OK-status: GpuIdManager::TfToCudaGpuId(tf_gpu_id, &cuda_gpu_id) status: Not found: TensorFlow device GPU:0 was not registered`
I use custom frozen graph to convert (2 x conv->3 x gru rnn->dense->softmax).
Other Tensorflow scripts (which are not using tensorrt) works correctly with both GPU.

### Source code / logs
**Source code:**

```
def load_graph(frozen_graph_filename):
    with tf.gfile.GFile(frozen_graph_filename, "rb") as f:
        graph_def = tf.GraphDef()
        graph_def.ParseFromString(f.read())
    return graph_def

graph_def = load_graph(model_path)

import tensorflow.contrib.tensorrt as trt2
trt_graph = trt2.create_inference_graph(
    input_graph_def=graph_def,
    outputs=['softmax_var/output'],
    max_batch_size=5,
    max_workspace_size_bytes=5 << 25,
    precision_mode='FP16',
    minimum_segment_size=1
)
with tf.Graph().as_default() as infer_graph:
    tf.import_graph_def(trt_graph, name="")
```

**Traceback:**
```
INFO:tensorflow:Running against TensorRT version 4.0.1
[INFO 2018-07-13 10:40:36,867 tf_logging.py:115] Running against TensorRT version 4.0.1
2018-07-13 10:40:38.625151: I tensorflow/core/grappler/devices.cc:51] Number of eligible GPUs (core count >= 8): 2
2018-07-13 10:40:41.120095: I tensorflow/contrib/tensorrt/convert/convert_graph.cc:747] MULTIPLE tensorrt candidate conversion: 428
2018-07-13 10:40:41.120688: I tensorflow/contrib/tensorrt/convert/convert_nodes.cc:2858] Segment @scope 'rnn/RNN_2/BGRU_2/bw/bw/while/custom_gru_cell/', converted to graph
<...many segment conversion messages...>
2018-07-13 10:41:02.943808: E tensorflow/contrib/tensorrt/convert/convert_graph.cc:711]  Can't find a GPU device to work with. Please instantiate a session to initialize devices
2018-07-13 10:41:02.943813: W tensorflow/contrib/tensorrt/convert/convert_graph.cc:833] Can't identify the cuda device. Running on device 0
2018-07-13 10:41:02.946091: W tensorflow/contrib/tensorrt/convert/convert_graph.cc:846] Engine creation for segment 108, composed of 1 nodes failed: Invalid argument: Output node 'rnn/RNN/BGRU_0/bw/bw/TensorArrayUnstack/strided_slice/stack_2' is weights not tensor. Skipping...
2018-07-13 10:41:02.946118: E tensorflow/contrib/tensorrt/convert/convert_graph.cc:711]  Can't find a GPU device to work with. Please instantiate a session to initialize devices
2018-07-13 10:41:02.946123: W tensorflow/contrib/tensorrt/convert/convert_graph.cc:833] Can't identify the cuda device. Running on device 0
2018-07-13 10:41:02.948399: W tensorflow/contrib/tensorrt/convert/convert_graph.cc:846] Engine creation for segment 109, composed of 1 nodes failed: Invalid argument: Output node 'rnn/RNN/BGRU_0/bw/custom_gru_cell/BatchNorm/moving_mean/read' is weights not tensor. Skipping...
2018-07-13 10:41:02.948426: E tensorflow/contrib/tensorrt/convert/convert_graph.cc:711]  Can't find a GPU device to work with. Please instantiate a session to initialize devices
2018-07-13 10:41:02.948430: W tensorflow/contrib/tensorrt/convert/convert_graph.cc:833] Can't identify the cuda device. Running on device 0
2018-07-13 10:41:02.950709: W tensorflow/contrib/tensorrt/convert/convert_graph.cc:846] Engine creation for segment 110, composed of 1 nodes failed: Invalid argument: Output node 'rnn/RNN_1/BGRU_1/fw/fw/ExpandDims/dim' is weights not tensor. Skipping...
2018-07-13 10:41:02.950719: W tensorflow/contrib/tensorrt/convert/convert_graph.cc:715] Can't determine the device, constructing an allocator at device 0
2018-07-13 10:41:02.950731: F ./tensorflow/core/common_runtime/gpu/gpu_id_utils.h:50] Non-OK-status: GpuIdManager::TfToCudaGpuId(tf_gpu_id, &cuda_gpu_id) status: Not found: TensorFlow device GPU:0 was not registered
Aborted (core dumped)

```

**Comment by qinyao-he** [comment]
I have also faced the same issue with PR #20350

**Comment by aaroey** [comment]
This is a bug in https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/tensorrt/convert/convert_graph.cc#L731, and will be fixed by #20318. Please wait for that PR to be merged and try again later.

Thanks.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @aaroey: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by aaroey** [comment]
I tested the script provided by @k-anastasia and it works well.

**Comment by qinyao-he** [comment]
@aaroey I still have the same issue with this PR merged. Although not core dumped, it end up with not changing the graph. The log information shows failed in TensorRT engine creation.

> 2018-08-07 15:19:10.464436: W tensorflow/contrib/tensorrt/convert/convert_graph.cc:814] Can't identify the cuda device. Running on device 0                                                        
> 2018-08-07 15:19:10.469900: E tensorflow/contrib/tensorrt/log/trt_logger.cc:38] DefaultLogger runtime.cpp (16) - Cuda Error in allocate: 2                                                         
> 2018-08-07 15:19:10.470166: E tensorflow/contrib/tensorrt/log/trt_logger.cc:38] DefaultLogger runtime.cpp (16) - Cuda Error in allocate: 2                                                         
> 2018-08-07 15:19:10.470190: W tensorflow/contrib/tensorrt/convert/convert_graph.cc:827] Engine creation for segment 1, composed of 22 nodes failed: Internal: Failed to build TensorRT engine. Skipping...

**Comment by aaroey** [comment]
@qinyao-he since it's different from this issue, would you please file a separate issue for that? 
Thanks.
